priv/docker.go

```diff
@@ -46,11 +46,15 @@ func (*unclosableConn) Close() error {
 
 // connectSockOpt overrides the client dialer to always return the same socket connection
 func connectSockOpt(host *url.URL) (client.Opt, error) {
-	socketConn, err := net.Dial("unix", host.Path)
-	if err != nil {
-		return nil, err
+	conns := make(chan *unclosableConn, 5)
+	for i := 0; i < 5; i++ {
```

Comment:
`len(conns)` (or even better, `runtime.NumCPU`?)

Reply:
outdated: moving in a different direction (preventing the need for multiple conns)